File: 17838.py
for i in range(int(input())):
    command = input()

    if len(command) != 7: print(0)
    else:
        if command[0] != command[1] or command[0] != command[4]:
            print(0)
        elif command[2] != command[3] or command[2] != command[5] or command[2] != command[6]:
            print(0)
        elif command[0] == command[2]:
            print(0)
        else:
            print(1)

File: 17838.cpp
#include <iostream>
using namespace std;

int main() {
	int T; cin >> T;

	for (int i = 0; i < T; i++) {
		string command; cin >> command;

		if (command.length() != 7) cout << 0 << endl;
		else {
			if (command[0] != command[1] || command[0] != command[4]) cout << 0 << endl;
			else if (command[2] != command[3] || command[2] != command[5] || command[2] != command[6])
				cout << 0 << endl;
			else if (command[0] == command[2]) cout << 0 << endl;
			else cout << 1 << endl;
		}
	}
}
